Sources/Sentry/SentryClient.m

```diff
@@ -366,6 +368,17 @@ - (SentryEvent *_Nullable)prepareEvent:(SentryEvent *)event
         event = self.options.beforeSend(event);
     }
 
+    // The callback is also called when SentrySDK.crashedLastRun is true and the
+    // SentryCrashReportSink didn't capture the crash event yet and the user manually captures an
+    // event with level fatal.
```

Comment:
M: `level.fatal` != `last program execution terminated with a crash.`

maybe we should only look to `exception.mechanism.handled=false`

Reply:
That's a good point. I'm going to check this out.